app/controllers/katello/api/v2/content_uploads_controller.rb

```diff
@@ -9,7 +9,7 @@ class Api::V2::ContentUploadsController < Api::V2::ApiController
     api :POST, "/repositories/:repository_id/content_uploads", N_("Create an upload request")
     param :repository_id, :number, :required => true, :desc => N_("repository id")
     def create
-      render :json => pulp_content.create_upload_request
+      render :json => Katello::Pulp::Content.create_upload
```

Comment:
I wonder if we should make a Repository#backend_content_service(smart_proxy) method to serve as the layer that determines which service class to use?

Reply:
I thought this was a good idea as well, but then I started thinking about Pulp 3 artifacts and how they aren't associated with a repository. We had talked previously about how Katello will associate artifacts with a repo anyway. I guess we'd just have to make sure we handle the case where the artifact is uploaded the second time for a second repo and already exists as an artifact in Pulp 3.

Reply:
sure, but i wonder if this is a problem today.  What happens if you upload the same rpm to two different repos?

Reply:
I am able to upload the same file to multiple repos and also the same repo multiple times, and pulp gladly accepts it. Seems like Pulp 3 has intentionally improved this behavior over Pulp 2.